url2698a2c731_cassandra_project_cassandra_stand_alone_tgz-pJ8-eu_cassandra_sim_MySimpleDynamicScenarioSimulationJ8/src/eu/cassandra/sim/math/response/Response.java

```diff
@@ -0,0 +1,669 @@
+
+						values[after] += additive;
+					}
+				}
+			}
+			// In case of reward a percentage of the close distances are moved in the
+			// window, either from one or both sides accordingly.
+			else {
+				side = Constants.SHIFTING_WINDOW_IN_MINUTES * 2;
+				switch (type) {
+
+				case "Both":
+
+					for (int i = startIndex - side; i < startIndex; i++) {
+
+						int index = i;
+
+						if (index < 0)
+							index += Constants.MINUTES_PER_DAY;
+
+						temp = incentive.getPrice() * values[index] / incentive.getBase();
+						overDiffTemp = (values[index] - temp) * awareness * sensitivity;
+						overDiff += overDiffTemp;
+						values[index] -= overDiffTemp;
+
+						// overDiff += values[index] - temp;
+						// values[index] = temp;
+					}
+
+					for (int i = endIndex; i < endIndex + side; i++) {
+
+						int index = i;
+
+						if (index > Constants.MINUTES_PER_DAY - 1)
+							index %= Constants.MINUTES_PER_DAY;
+
+						temp = incentive.getPrice() * values[index] / incentive.getBase();
+						overDiffTemp = (values[index] - temp) * awareness * sensitivity;
+						overDiff += overDiffTemp;
+						values[index] -= overDiffTemp;
+
+						// temp = incentive.getPrice() * values[index] /
+						// incentive.getBase();
+						// overDiff += values[index] - temp;
+						// values[index] = temp;
+					}
+					break;
+
+				case "Left":
+
+					for (int i = startIndex - 2 * side; i < startIndex; i++) {
+
+						int index = i;
+
+						if (index < 0)
+							index += Constants.MINUTES_PER_DAY;
+
+						temp = incentive.getPrice() * values[index] / incentive.getBase();
+						overDiffTemp = (values[index] - temp) * awareness * sensitivity;
+						overDiff += overDiffTemp;
+						values[index] -= overDiffTemp;
+
+						// temp = incentive.getPrice() * values[index] /
+						// incentive.getBase();
+						// overDiff += values[index] - temp;
+						// values[index] = temp;
+					}
+					break;
+
+				case "Right":
+
+					for (int i = endIndex; i < endIndex + 2 * side; i++) {
+
+						int index = i;
+
+						if (index > Constants.MINUTES_PER_DAY - 1)
+							index %= Constants.MINUTES_PER_DAY;
+
+						temp = incentive.getPrice() * values[index] / incentive.getBase();
+						overDiffTemp = (values[index] - temp) * awareness * sensitivity;
+						overDiff += overDiffTemp;
+						values[index] -= overDiffTemp;
+
+						// temp = incentive.getPrice() * values[index] /
+						// incentive.getBase();
+						// overDiff += values[index] - temp;
+						// values[index] = temp;
+					}
+
+				}
+				// System.out.println("Over Difference = " + overDiff);
+
+				double additive = overDiff / (endIndex - startIndex);
+
+				for (int i = startIndex; i < endIndex; i++)
+					values[i] += additive;
+
+			}
+
+		}
+		// for (int i = 0; i < values.length; i++)
+		// sum += values[i];
+		// System.out.println("Summary: " + sum);
+
+		return values;
+	}
+
+	private static double[] discreteOptimal(double[] values, PricingVector pricing,
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 16. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.